File: atcoder/abc390b_geometric_sequence.py
# Vicfred
# https://atcoder.jp/contests/abc390/tasks/abc390_b
# math, implementation
import sys

N = int(input())
# print(N)
A = list(map(int, input().split()))
#print(A)
if len(A) <= 2:
    print("Yes")
    sys.exit()
ratio = A[1] / A[0]
valid = True
#print("ratio:", ratio)
for x in range (len(A) - 2):
    if A[x] * A[x + 2] != A[x + 1] * A[x + 1]:
        valid = False
        #print("failed", A[x] * ratio, A[x + 1])
        break
if valid:
    print("Yes")
else:
    print("No")

File: atcoder/abc390b_geometric_sequence.rs
// Vicfred
// https://atcoder.jp/contests/abc390/tasks/abc390_b
// math, implementation
use std::io;

fn main() {
    let mut line1 = String::new();

    io::stdin().read_line(&mut line1).unwrap();

    let N : i64 = line1.trim().parse().unwrap();

    let mut line2 = String::new();

    io::stdin().read_line(&mut line2).unwrap();

    let mut A : Vec<i64> =
        line2
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    let mut valid : bool = true;
    for i in 0..(N-2) as usize {
        if A[i] * A[i + 2] != A[i + 1] * A[i + 1] {
            valid = false;
            break;
        }
    }
    if valid {
        println!("Yes");
    }
    else {
        println!("No");
    }
}
